milatools/utils/vscode_utils.py

```diff
@@ -192,38 +187,36 @@ def install_vscode_extensions_task_function(
        extensions on the source;
     4. Install the extensions that are missing or out of date on the remote, updating
        the progress dict as it goes.
+
+
+    Returns the list of installed extensions, in the form 'extension_name@version'.
     """
+    installed: list[str] = []
 
     def _update_progress(
         progress: int, status: str, total: int = len(source_extensions)
     ):
-        # Show progress to the parent process by setting an item in the task progress
-        # dict.
-        progress_dict: ProgressDict = {
-            "progress": progress,
-            "total": total,
-            "info": textwrap.shorten(status, 50, placeholder="..."),
-        }
-        task_progress_dict[task_id] = progress_dict
-        return progress_dict
-
-    if remote is None:
+        info = textwrap.shorten(status, 50, placeholder="...")
+        report_progress(progress=progress, total=total, info=info)
+
+    if not remote:
         if dest_hostname == "localhost":
-            remote = LocalV1()
+            remote = LocalV2()
```

Comment:
I'm not sure to understand how and why we would be syncing to localhost? Wouldn't it be simpler to manage the extensions locally through the vscode interface?

Reply:
You might want to get the same vscode extensions locally as you had on the Mila cluster. Sure, if you have "settings sync" setup in vscode, it could probably sync it better than this would, but I don't see why we shouldn't support doing this.